lint-workflow/lint.py

```diff
@@ -292,9 +295,24 @@ def lint(filename):
                             logging.info("Skipping local action in workflow.")
                             break
 
-                        # If the step has a 'uses' key, check value hash, except bitwarden actions.
+                        # If the step has a 'uses' key, check if actions are in supported actions list and also value hash, except bitwarden actions.
                         if "bitwarden/gh-actions" not in path:
                             try:
+                                # Check if actions are in supported actions list.
+                                actions_count = 0
```

Comment:
What's the BigO notation of this approach?

Reply:
Best-case would be `n` worst case would be `n^2`... it will check the actions in steps individually,

Reply:
How might we be able to improve this down to a constant instead of `n`?

Reply:
@joseph-flinn I have updated the data structure and how I'm comparing the actions, please review

Reply:
Nice! We have it down to a constant 🙌! 

Hashmaps are great for looking up information (or `value`) based on a specific `key`. What other data structure gets a constant lookup and is more suited to checking for existence vs looking up data that could be unrelated?

Reply:
HashSet, oh nice, I will use HashSet. in place of map

Reply:
Awesome! I've approved the PR. Thanks!